src/spec/create_extension_spec.py

```diff
@@ -1,47 +1,394 @@
+                                       neurodata_type_def='UserCommentDevice',
+                                       neurodata_type_inc='Device',
+                                       datasets=[NWBDatasetSpec('device specific user text notes',
+                                                                neurodata_type_inc='UserCommentString')
+                                                 ]
+                                       )
+
+    user_comment = NWBGroupSpec(doc='Free form text notes from the experimenter',
+                                neurodata_type_def='UserComment',
+                                groups=[NWBGroupSpec('Device for the user text notes',
+                                                     neurodata_type_inc='UserCommentDevice',
+                                                     datasets=[NWBDatasetSpec('device specific user text notes',
+                                                                              neurodata_type_inc='UserCommentString'
+                                                                              ),
+                                                               ]
+                                                     )
+                                        ]
+                                )
+
+    generated_by = NWBDatasetSpec(doc='Software provenance information as key '
+                                  '(first column) value (second column) pairs.',
+                                  neurodata_type_def='GeneratedBy',
+                                  neurodata_type_inc='NWBData',
+                                  shape=(None, 2))
+
+    mies_general = NWBGroupSpec(doc='Additional data and metadata from MIES',
+                                neurodata_type_def='MIESMetaData',
+                                neurodata_type_inc='LabMetaData',
+                                datasets=[NWBDatasetSpec(doc='Software provenance information as key '
+                                                         '(first column) value (second column) pairs.', neurodata_type_inc='GeneratedBy')],
+                                groups=[NWBGroupSpec(doc='Free form text notes from the experimenter',
+                                                     neurodata_type_inc='UserComment',
+                                                     groups=[NWBGroupSpec('Device for the user text notes',
+                                                                          neurodata_type_inc='UserCommentDevice',
+                                                                          datasets=[NWBDatasetSpec('device specific user text notes',
+                                                                                                   neurodata_type_inc='UserCommentString')]
+                                                                          )
+                                                             ]
```

Comment:
These lines are already in the NWBGroupSpec definition for the neurodata_type UserComment in line 249.
```suggestion
```